apps/pixeloven/site/docs/recommendations/react-components.mdx

```diff
@@ -75,4 +75,53 @@ ADD LINK TO DOCUMENTATION ON ROUTES
 
 ## Component composition
 
-ADD DOCUMENTATION ON COMPONENT COMPOSITION
+When it comes to extracting components, there are two important factors to consider:
+1. complexity of code (e.g. conditional rendering and other business logic)
+2. reusability of code
+
+We use the atomic design as our philosophy for creating reusable React components, and create a new storybook entry for each new atomic design component. However, sometimes an atomic design component itself might include complex business logic for rendering, or code that is reused only within that component, and extracting some of its pieces into their own components would help improve readability and maintainability of the code. If you see that these pieces need to be extracted into their own component, yet they do not warrant creating a new atomic design component in storybook, then simply create new React component files directly within the same storybook folder.
```

Comment:
We use the atomic design as our philosophy for creating reusable React components, and create a new storybook entry for each new atomic design component.

Atomic design is redundant here and since it's not just storybook but individual tests and style files as well, what do you think about udpating this to

We use the Atomic Design as our philosophy for creating reusable React components, along with additional co-located support files for each new component.

Reply:
> If you see that these pieces need to be extracted into their own component, yet they do not warrant creating a new atomic design component in storybook, then simply create new React component files directly within the same storybook folder.

Similarly here, what do you think about updating this to 

> If you see that these pieces need to be extracted into their own component, yet they do not warrant creating the additional supporting files, then simply create a co-located React component within the original component folder and import it into the original component. Any necessary styles, stories, or tests then live within the original support files.

Reply:
Updated - [commit](https://github.com/pixeloven/pixeloven/pull/288/commits/384b724df613ba9b5930f86f41b9ab6a9614d8d4)